x-pack/plugins/endpoint/common/types.ts

```diff
@@ -25,35 +25,39 @@ export type ImmutableObject<T> = { readonly [K in keyof T]: Immutable<T[K]> };
 export class EndpointAppConstants {
   static ALERT_INDEX_NAME = 'my-index';
   static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
-}
-
-export interface AlertResultList {
+  static EVENT_INDEX_NAME = 'endpoint-events-*';
   /**
-   * The alerts restricted by page size.
+   * Phase 0 events are stored in indices with endgame-* prefix
    */
-  alerts: AlertData[];
+  static ENDGAME_INDEX_NAME = 'endgame-*';
+}
 
+export interface Pagination {
   /**
-   * The total number of alerts on the page.
+   * The total number of items on the page.
    */
   total: number;
-
   /**
    * The size of the requested page.
    */
   request_page_size: number;
-
   /**
    * The index of the requested page, starting at 0.
    */
   request_page_index: number;
-
   /**
    * The offset of the requested page, starting at 0.
    */
   result_from_index: number;
 }
 
+export interface AlertResultList extends Pagination {
+  /**
+   * The alerts restricted by page size.
+   */
+  alerts: AlertData[];
+}
+
```

Comment:
`EndpointResultList` can eventually use the `Pagination` interface too once it's returning the `result_from_index` field.

Reply:
why inheritance? I think the list should have pagination properties and should not be a pagination.

Reply:
This was just to reduce code duplication. Yeah that's true that composition would be better, but to do that our results will need to be structured like this:

```
interface AlertResultList {
...
pagination: Pagination;
}
```

So I was trying to avoid breaking changes with how the results are formatted right now. Another alternative to `extends` is `AlertResultList & Pagination` but that looks like of ugly to me:
https://stackoverflow.com/a/49723584/12528274

Reply:
Also this: https://www.typescriptlang.org/docs/handbook/interfaces.html#extending-interfaces

Reply:
hmm I see the angle, but AlertResultList is not a pagination from an inheritance point of view. BaseList make more sense if you want to go the inheritance route.

Reply:
Ah ok, how about `BaseResult`? Resolver's response is sort of a list I guess. `Results` seems a bit more generic?

Reply:
That works, thanks for the update. We just have to be wary of deep inheritance.

Reply:
I like BaseResult 👍